File: go/callee-service-gofr/logic/calleelogic.go
package logic

import "callee-service/controller/dto"

func SayMyName(name string) dto.Callee {
	return dto.Callee{ID: "0", Message: "Your name is: " + name}
}

func SayMyOtherName(name string) dto.Callee {
	return dto.Callee{ID: "0", Message: "Your other name is: " + name}
}

func Save(callee dto.Callee) dto.Callee {
	return dto.Callee{ID: "0", Message: "Storing your message: " + callee.Message}
}

File: net/callee-service-net/src/CalleeService/Logic/CalleeLogic.cs
using GoaFabric.CalleeService.Models;

namespace GoaFabric.CalleeService.Logic;

/// <summary>
/// Business logic — equivalent of the Quarkus @ApplicationScoped CalleeLogic bean.
/// Registered as a singleton in Program.cs (services.AddSingleton).
/// </summary>
public class CalleeLogic
{
    public Callee SayMyName(string name)
        => new("0", $"Your name is: {name}");

    public Callee SayMyOtherName(string name)
        => new("0", $"Your other name is: {name}");

    public Callee Save(Callee callee)
        => new("0", $"Storing your message: {callee.Message}");
}
